### PR TITLE
Feature: Input and output vector potential in tddft calculation

### PR DESCRIPTION
**Pull request by ESROAMER**
### Linked Issue
Add feature requested by issue #3845 

### What's changed?
I add some function and two input parameters to allow the user to init vector potential A(t) from file "At.dat" or to output A(t) into file "OUT.suffix/At.dat" .

### CODE REVIEW

File: source/module_hamilt_lcao/module_tddft/td_velocity.cpp [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+          << std::setw(15) << "A_z" << std::endl;
+    }
+    else
+    {
+      ofs.open(out_file.c_str(), std::ofstream::app);
+    }
+    //output the vector potential
+    ofs << std::left <<std::setw(8) << istep;
+    // divide by 2.0 to get the atomic unit
+    for(int i=0; i<3; i++)
```
Comment (mohanchen) [inline source/module_hamilt_lcao/module_tddft/td_velocity.cpp]: add {} after all 'for' loop and 'if'

File: source/module_hamilt_lcao/module_tddft/td_velocity.cpp [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+      ofs << std::scientific << std::setprecision(4)<< std::setw(15) << cart_At[i]/2.0;
+    ofs << std::endl;
+    ofs.close();
+  }
+  return;
+}
+
+void TD_Velocity::cal_cart_At(ModuleBase::Vector3<double> a0,
```
Comment (mohanchen) [inline source/module_hamilt_lcao/module_tddft/td_velocity.cpp]: use 'const ModuleBase::Vector3<double> &a0' instead?

File: source/module_hamilt_lcao/module_tddft/td_velocity.cpp [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+                              ModuleBase::Vector3<double> At)
+{
+  istep++;
+  if(init_vecpot_file)
+  {
+    this->cart_At=At_from_file[istep>max_istep?max_istep:istep];
+  }
+  else
+  {
+    double l_norm[3]={a0.norm() ,a1.norm() ,a2.norm()};
```
Comment (mohanchen) [inline source/module_hamilt_lcao/module_tddft/td_velocity.cpp]: add 'const' before 'double'?

File: source/module_hamilt_lcao/module_tddft/td_velocity.cpp [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+    At*=2.0;
+    //add the tmporary vector3 to the vector potential vector
+    At_from_file.push_back(At);
+  }
+  //set the max_istep
+  max_istep = At_from_file.size()-1;
+  ifs.close();
+
+  return;
+}
```
Comment (mohanchen) [inline source/module_hamilt_lcao/module_tddft/td_velocity.cpp]: Did you use 'clang_format' for all of the files?

Comment (ESROAMER) [reply source/module_hamilt_lcao/module_tddft/td_velocity.cpp]: Sorry, I don't know what it is. Could you please tell me more about it? @mohanchen

File: source/module_hamilt_lcao/module_tddft/td_velocity.cpp [hunk 5]
```diff
@@ -1,24 +1,144 @@
...
+        {
+            ofs << std::scientific << std::setprecision(4) << std::setw(15) << cart_At[i] / 2.0;
+        }
+        ofs << std::endl;
+        ofs.close();
+    }
+    return;
+}
+
+void TD_Velocity::cal_cart_At(const ModuleBase::Vector3<double> a0,
```
Comment (mohanchen) [inline source/module_hamilt_lcao/module_tddft/td_velocity.cpp]: add "&" before a0, a1, a2, and a3

File: source/module_hamilt_lcao/module_tddft/td_velocity.cpp [hunk 6]
```diff
@@ -1,24 +1,144 @@
...
+    // output the vector potential if needed
+    if (out_vecpot == true)
+    {
+        this->output_cart_At(GlobalV::global_out_dir);
+    }
+}
+
+void TD_Velocity::read_cart_At(void)
+{
+
```
Comment (mohanchen) [inline source/module_hamilt_lcao/module_tddft/td_velocity.cpp]: delete the extra blank line

File: source/module_hamilt_lcao/module_tddft/td_velocity.cpp [hunk 7]
```diff
@@ -1,24 +1,144 @@
...
+    }
+    std::string line;
+    std::vector<std::string> str_vec;
+    // use tmp to skip the istep number
+    int tmp = 0;
+    while (std::getline(ifs, line))
+    {
+        // A tmporary vector3 to store the data of this line
+        ModuleBase::Vector3<double> At;
+        if (line[0] == '#')
```
Comment (mohanchen) [inline source/module_hamilt_lcao/module_tddft/td_velocity.cpp:103]: add {} after if

File: source/module_hamilt_lcao/module_tddft/td_velocity.cpp [hunk 8]
```diff
@@ -1,24 +1,144 @@
...
+            continue;
+        std::istringstream iss(line);
+        // skip the istep number
+        if (!(iss >> tmp))
+        {
+            ModuleBase::WARNING_QUIT("TD_Velocity::read_cart_At", "Error reading istep!");
+        }
+        // read the vector potential
+        double component;
```
Comment (mohanchen) [inline source/module_hamilt_lcao/module_tddft/td_velocity.cpp]: initialize the variable

**Review by mohanchen — APPROVED** [review]

**Pull request merged**